### DISCUSSION TITLE
Delete Resource

### DISCUSSION THREAD

**Discussion by JamesDolden** [filamentphp/filament#1864] [Help] [answered]
Hi, 

Can someone please confirm the best way to completely delete a resource that has been created? Just deleting the files seems to cause errors when trying to generate new resources?

Thanks, 

  James

**Comment by ryangjchandler** [comment, marked as answer]
Deleting the `*Resource.php` file and `Resources/*Resource` folder should work. What error are you getting?

**Reply by JamesDolden** [reply]
So I have a resource called CheckoutResource I need to delete. I delete the file and folder as you mention. The resource remains on the sidebar of /admin and then also if I try and create a new resource, I get:

`include(/var/www/html/app/Filament/Resources/CheckoutResource.php): Failed to open stream: No such file or directory`

**Reply by JamesDolden** [reply]
OK, so I am using Laravel Sail, and just tried closing and reloading Sail after deleting the files and that seems to have done the trick.

**Reply by ryangjchandler** [reply]
Yeah, you either need to restart the server with Sail or run a `composer dump-autoload` I believe.

**Comment by Greatwolf3** [comment]
just run the following command php artisan filament:clear-cached-components

**Reply by itsmejyv** [reply]
Thanks for this, it helped me to debug the error that was related to this.
